lib/process/master.js

```diff
@@ -75,6 +75,17 @@ process.on('message', function(msg) {
   }
 });
 
+process.on('uncaughtException', function(err) {
+  if (!err.message) {
+    err = new Error(err);
+  }
+  process.send({
+    cmd: 'failed',
+    value: err
+  });
+  throw err;
```

Comment:
is this throw needed?

Reply:
I would say so, if there's an uncaught exception, we have no guarentees as to the state of the worker so it's best to just let it crash after we pass the error up to the parent